control.lua

```diff
@@ -49,7 +49,12 @@ script.on_event({defines.events.on_console_chat},
 
 script.on_event(defines.events.on_player_died, function(event)
 	local p = game.players[event.player_index];
-	FactoCordIntegration.PrintToDiscord("**" .. p.name .. "** died.");
+	local c = event.cause
+	if not c then
+		FactoCordIntegration.PrintToDiscord("**" .. p.name .. "** died.");
+	else
+		FactoCordIntegration.PrintToDiscord({"", "**" .. p.name .. "** died to a " , c.localised_name, "."});
```

Comment:
The first thing I tried was `c.name` but that threw an error because the name property doesn't seem exist on the few entities I tested (biters and worms).  
However localised_name existed. So I tried that and got the error can't concatenate a table and string.


For example, say I die to a small biter then localised_name will be the table `{"small-biter"}`. So in order to get what is desired I have to use functions that can/will handle a LocalisedString. Then in order to keep everything as a LocalisedString the whole way through I have to do the LocalisedString version of string concatenation. Then due to determinism I don't have a way to resolve the LocalisedString to a plain string on demand (well there is the translate function that later creates an event with the results but that is a lot of unneeded delays and complications, IMO). So I have to do it this way so that Factorio's C/C++ code generates the plain string from the requested LocalisedString outside of the GameState and therefore doesn't affect the determinism. Really hope I explained that decently/correctly.

For some reason all functions will accept LocalisedString except the global `print()` function. Instead, there is a different global function named `localised_print()` that will accept a LocalisedString. I'm sure there is a reason why this was done but I certainly don't know it. Best guess is changing the lua's global built-in print() function is a bad idea.

Reply:
Yes, now it seems reasonable